### ISSUE TITLE
[Counter-Strike] Oportunity to block VGUI Buy Menu from server side

### ISSUE THREAD

**Issue by 6y6JIuK** [ValveSoftware/halflife#2288]
If i want to block original buy menu (vgui style), i need to block client command(to block old style buymenu) and send message BuyClose(to close vgui bm)

But when you send buyclose its needs some time to be done and player can see that menu.
So it is seriously dependes by player's latency. You can see it on the video below. If player ping more then 60, time for closing bm will increase

https://youtu.be/Pw0ppO9G3Uc

**Comment by SamVanheer** [comment]
Why do you need to block the menu?

**Comment by eat1k** [comment]
> Why do you need to block the menu?

I think to show correctly a custom one. It would be very nice to block completely VGUI Buy.

**Comment by 6y6JIuK** [comment]
@SamVanheer  custom buymenu
I have zp mod, when you wanna get some items from it, vgui menu can be problematic in some critical situations in the gameplay, because 1) it can lock player control for few miliseconds; 2) it is creating delay before custom menu will be deployed and can confuse the player when he trying to get items(cant do fast keys combinations, like b-2-1 for example)

Player can avoid it by setting the cvar _vgui_menu 0, but its cannot be done by server by some reasons

**Comment by a1batross** [comment]
The difference between VGUI menus and old styled ones are on server side completely. VGUI menus are sent with separate VGUIMenu message and old styled just ShowMenu.

https://github.com/s1lentq/ReGameDLL_CS/blob/master/regamedll/dlls/player.cpp#L8171
You can just ignore the user's value and send old styled. Don't see any problem here.

**Comment by voed** [comment]
> The difference between VGUI menus and old styled ones are on server side completely. VGUI menus are sent with separate VGUIMenu message and old styled just ShowMenu.
> 
> https://github.com/s1lentq/ReGameDLL_CS/blob/master/regamedll/dlls/player.cpp#L8171
> You can just ignore the user's value and send old styled. Don't see any problem here.

VGUI menus has client prediction and cannot be blocked completely, as i member

**Comment by a1batross** [comment]
@voed no, even _vgui_menus cvar isn't used in client.dll code. It exists only on one purpose: to be sent on a server in user info buffer.

**Comment by 6y6JIuK** [comment]
@a1batross 
try to block it byself
just check it

**Comment by shelru** [comment]
And what's wrong with BuyClose?

**Comment by voed** [comment]
@shelru it can be sent only after VGUI already opened, so user will see flicker with VGUI menu

**Comment by shelru** [comment]
Вряд-ли это можно отнести к багам. Ведь мы понимаем каким образом формируется закупочное меню и почему оно на sside, а вот автор ишью - нет.

**Comment by a1batross** [comment]
Yeah, indeed, I was confused with my own client.dll, which doesn't have this flaw.
